here-naksha-lib-model/src/commonMain/kotlin/naksha/model/NakshaLockProxy.kt

```diff
@@ -18,40 +22,22 @@ import kotlin.jvm.JvmStatic
  * Another approach would be to use HERE tile-ids as lock-ids.
  */
 @JsExport
-class P_NakshaLock : GeoFeature() {
+class NakshaLockProxy : NakshaFeatureProxy() {
 
     companion object {
-        @JvmStatic
-        val TYPE = Platform.intern("type")
-
-        @JvmStatic
-        val MAX_OWNERS = Platform.intern("maxOwners")
-
-        @JvmStatic
-        val OWNER = Platform.intern("owner")
-
-        @JvmStatic
-        val EXPIRES = Platform.intern("expires")
-
-        @JvmStatic
-        val OWNERS = Platform.intern("owners")
+        private val TYPE = NotNullProperty<Any, NakshaLockProxy, String>(String::class, defaultValue = "naksha.Lock")
+        private val MAX_OWNERS = NotNullProperty<Any, NakshaLockProxy, Int>(Int::class, defaultValue = 1)
+        private val OWNER = NullableProperty<Any, NakshaLockProxy, String>(String::class)
+        private val EXPIRES = NotNullProperty<Any, NakshaLockProxy, Int64>(Int64::class)
+        private val OWNERS = NullableProperty<Any, NakshaLockProxy, NakshaLockProxy.Owners>(
+            NakshaLockProxy.Owners::class)
     }
 
-    fun getType(): String = "naksha.Lock"
-
-    fun getMaxOwners(): Int = getAs(MAX_OWNERS, Platform.intKlass, 1)
-
-    fun setMaxOwners(maxOwners: Int) = set(MAX_OWNERS, maxOwners)
-
-    fun getOwner(): String? = getOrNull(OWNER, Platform.stringKlass)
-
-    fun setOwner(value: String?) = set(OWNER, value)
-
-    fun setExpires(value: Int64?) = set(EXPIRES, value)
-
-    fun getExpires(): Int64? = getOrNull(EXPIRES, Platform.int64Klass)
-
-    fun setOwners(value: P_Map<String, Int64>?) = set(OWNERS, value)
+    var type: String by TYPE
+    var maxOwners: Int by MAX_OWNERS
+    var owner: String? by OWNER
+    var expires: Int64 by EXPIRES
+    var owners: NakshaLockProxy.Owners? by OWNERS
 
-    fun getOwners(): P_Map<String, Int64>? = getOrNull(OWNERS, P_Map::class) as P_Map<String, Int64>?
+    class Owners : P_Map<String, Int64>(String::class, Int64::class)
```

Comment:
Can move this class into an own file and export to JavaScript (e.g. `NakshaLockOwnersProxy`)?